phpBB/phpbb/install/module/update_filesystem/task/file_check.php

```diff
@@ -138,6 +138,15 @@ function ($filename) use ($root_path)
 			$progress_count++;
 			$this->iohandler->set_progress('UPDATE_CHECK_FILES', $progress_count);
 
+			// Do not copy viglink again if the previous version was packaged
+			// with it but it does not exist (e.g. deleted by admin)
+			if (strpos($file, $this->phpbb_root_path . 'ext/phpbb/viglink') !== false &&
+				$this->update_helper->phpbb_version_compare($update_info['version']['from'], '3.2.0', '>=') &&
+				!$this->filesystem->exists($this->phpbb_root_path . 'ext/phpbb/viglink'))
```

Comment:
Why would that skip a file if it's a new file in viglink? It will however also copy / compare files even if composer.json does not exist, which means it might update files if only the folder is left. Checking for composer.json is probably better.